sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -187,6 +187,23 @@ object EliminateOuterJoin extends Rule[LogicalPlan] with PredicateHelper {
   }
 }
 
+/**
+ * Convert inner join to left semi join.
+ */
+object EliminateInnerJoin extends Rule[LogicalPlan] with PredicateHelper with JoinSelectionHelper {
+  def apply(plan: LogicalPlan): LogicalPlan = {
+    plan.transformWithPruning(_.containsPattern(INNER_LIKE_JOIN), ruleId) {
+      case p @ Project(_,
+          j @ ExtractEquiJoinKeys(Inner, _, rightKeys, None, _, left, right: Aggregate, _))
```

Comment:
Inner join is symmetrical, can this rule also be applied to inner join with left `Aggregate` child?

`j @ ExtractEquiJoinKeys(Inner, leftKeys, _, _, None, left: Aggregate, _, _))`